src/openrct2/network/NetworkServerAdvertiser.cpp

```diff
@@ -229,14 +226,18 @@ class NetworkServerAdvertiser : public INetworkServerAdvertiser
     {
         // Generate a string of 16 random hex characters (64-integer key as a hex formatted string)
         static const char hexChars[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
-        char key[17];
-        for (sint32 i = 0; i < 16; i++)
+        char * key = new char[TWITCH_KEY_LENGTH + 1];
+
+        for (sint32 i = 0; i < TWITCH_KEY_LENGTH; i++)
         {
             sint32 hexCharIndex = util_rand() % Util::CountOf(hexChars);
             key[i] = hexChars[hexCharIndex];
         }
-        key[Util::CountOf(key) - 1] = 0;
-        return key;
+        key[TWITCH_KEY_LENGTH] = 0;
+
+        std::string result { key };
+        delete[] key;
+        return result;
```

Comment:
This seems good, but maybe deserves a separate commit?

Reply:
Its probably better to use more of std::unique_ptr to begin with.

Reply:
Ah yes, this should be in the PR. Must've missed this while splitting them up